objectRecognitionSource/src/main/java/edu/ml/tensorflow/util/ImageUtil.java

```diff
@@ -63,7 +64,33 @@ public void saveImage(final BufferedImage image, final String target) {
         if (outputDir != null) {
             try {
                 IOUtil.createDirIfNotExists(new File(outputDir));
-                ImageIO.write(image,"jpg", new File(outputDir + File.separator + target));
+                if (longEdge == 0) {
+                    ImageIO.write(image, "jpg", new File(outputDir + File.separator + target));
+                } else {
+                    int width = image.getWidth();
+                    int height = image.getHeight();
+                    if (longEdge > Math.max(width, height)) {
+                        LOGGER.error("The longEdge value is too large, resizing cannot enlarge the original image. "
```

Comment:
I think we should make this a trace() level note.  We've fulfilled our contract (see comment above on the readme) so there's nothing amiss here.